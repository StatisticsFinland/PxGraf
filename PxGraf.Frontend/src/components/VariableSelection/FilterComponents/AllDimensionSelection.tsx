import React from 'react';

import { Box, Typography } from '@mui/material';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';

const ComponentWrapper = styled(Box)`
  align-self: stretch;
  display: flex;
  align-items: center;
`;

export const AllDimensionSelection: React.FC = () => {
    const { t } = useTranslation();
    return (<ComponentWrapper><Typography>{t("variableSelect.allFilter")}</Typography></ComponentWrapper>);
}

export default AllDimensionSelection;