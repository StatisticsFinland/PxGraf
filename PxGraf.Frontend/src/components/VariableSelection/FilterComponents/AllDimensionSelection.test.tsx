import React from 'react';
import { render } from "@testing-library/react";
import AllDimensionSelection from "./AllDimensionSelection";

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<AllDimensionSelection
        ></AllDimensionSelection>);
        expect(asFragment()).toMatchSnapshot();
    });
});