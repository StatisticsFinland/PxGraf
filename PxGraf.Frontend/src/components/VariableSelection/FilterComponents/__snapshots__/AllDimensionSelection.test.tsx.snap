// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-Qotzb hUwrYH MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      variableSelect.allFilter
    </p>
  </div>
</DocumentFragment>
`;
